game-core/src/main/java/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -165,129 +163,100 @@ private void createAndSaveStats(final boolean showPhaseStats) {
         for (final UnitType unitType : gameData.getUnitTypeList()) {
           final UnitAttachment ua = UnitAttachment.get(unitType);
           if (ua == null) {
             continue;
           }
-          String toModify = ua.allUnitStatsForExporter();
-          toModify = toModify.replaceFirst("UnitType called ", "").replaceFirst(" with:", "")
-              .replaceAll("games.strategy.engine.data.", "").replaceAll("\n", ";").replaceAll(",", ";");
-          toModify = toModify.replaceAll("  ", ",");
-          toModify = toModify.replaceAll(", ", ",").replaceAll(" ,", ",");
-          text.append(toModify);
-          text.append("\n");
+          final String toModify = ua.allUnitStatsForExporter()
+              .replaceAll("UnitType called | with:|games.strategy.engine.data.", "")
+              .replaceAll("[\n,]", ";")
+              .replaceAll(" {2}| ?, ?", ",");
+          text.append(toModify).append('\n');
         }
       }
-      text.append("\n");
-      text.append((showPhaseStats ? "Full Stats (includes each phase that had activity),"
-          : "Short Stats (only shows first phase with activity per player per round),"));
-      text.append("\n");
-      text.append("Turn Stats: ,");
-      text.append("\n");
+      text.append('\n');
+      text.append(showPhaseStats ? "Full Stats (includes each phase that had activity),"
+          : "Short Stats (only shows first phase with activity per player per round),").append('\n');
+      text.append("Turn Stats: ,").append('\n');
       text.append("Round,Player Turn,Phase Name,");
-      for (final IStat stat : stats) {
-        for (final PlayerId player : players) {
-          text.append(stat.getName()).append(" ");
-          text.append(player.getName());
-          text.append(",");
-        }
-        for (final String alliance : alliances) {
-          text.append(stat.getName()).append(" ");
-          text.append(alliance);
-          text.append(",");
-        }
-      }
-      for (final IStat element : statsExtended) {
-        for (final PlayerId player : players) {
-          text.append(element.getName()).append(" ");
-          text.append(player.getName());
-          text.append(",");
-        }
-        for (final String alliance : alliances) {
-          text.append(element.getName()).append(" ");
-          text.append(alliance);
-          text.append(",");
+      final String[] alliances = statPanel.getAlliances().toArray(new String[0]);
+      final PlayerId[] players = statPanel.getPlayers().toArray(new PlayerId[0]);
+      // its important here to translate the player objects into our game data
+      // the players for the stat panel are only relevant with respect to the game data they belong to
+      Arrays.setAll(players, i -> clone.getPlayerList().getPlayerId(players[i].getName()));
+      for (final IStat[] statArray : Arrays.asList(stats, statsExtended)) {
+        for (final IStat stat : statArray) {
```

Comment:
It might be simpler to collapse these two loops into one using Guava's `Iterables#concat()`:

```java
for (final IStat stat : Iterables.concat(Arrays.asList(stats), Arrays.asList(statsExtended))) {
```

Reply:
Excellent suggestion 👍 
I checked the guava source a new iterator is being created for every call to iterator() so we are safe.